1-BasicSetUp.sh

```diff
@@ -67,15 +67,16 @@ if [[ -d $zsh_folder ]];then
 fi
 
 spatialPrint "Setting up Zsh + Zim now"
-execute sudo apt-get install zsh -y
+# Ubuntu 16.04 has older zsh in repository
+if [[ $(cat /etc/os-release | grep "VERSION_ID" | grep -o -E '[0-9][0-9]' | head -n 1) -le 16 ]]; then
+    sh -c "$(wget https://gist.githubusercontent.com/rsnk96/87229bd910e01f2ee7c35f96d7cb2f6c/raw/f068812ebd711ed01ebc4c128c8624730ab0dc81/build-zsh.sh -O -)"
```

Comment:
Might be better to stick to a release version rather than latest master 
MAybe even the version used by 18.04 (5.4.2)

Reply:
Would also have to lock `zimfw` and other `zsh` modules from updating if this is to be implemented, might make things slightly complicated....

Reply:
So specific versions of `zimfw` and `zsh` both?

Reply:
Need to lock zimfw updating too

Reply:
But we can leave it for now, I think that would be solving a problem which does not yet exist...

Reply:
I think we should do it, both `zsh` and `zimfw` versions should be fixed to be ones tested to be working, or at least going by master for both doesn't seem good

Reply:
Right now, `zsh` on source build is the latest release, as for `zimfw`, the script downloads the latest release for that as well, can't change that, other way would be manual install which is complex, and can change easily

Reply:
Hmmm agreed. Disabling zimfw updates is detailed [here](https://github.com/zimfw/zimfw#settings)

Reply:
Well, yes, we could disable any updates, but maybe that should be upto the user?